spec/services/billable_metrics/prorated_aggregations/unique_count_service_spec.rb

```diff
@@ -416,4 +416,89 @@
       end
     end
   end
+
+  describe '.per_event_aggregation' do
+    context 'with event added in the period' do
+      let(:added_at) { from_datetime + 10.days }
+
+      it 'aggregates per events' do
+        result = unique_count_service.per_event_aggregation
+
+        expect(result.event_aggregation).to eq([1])
+        expect(result.event_prorated_aggregation.map { |el| el.ceil(5) }).to eq([21.fdiv(31).ceil(5)])
+      end
+    end
+
+    context 'with persisted metrics removed in the period' do
+      let(:removed_at) { to_datetime - 15.days }
+
+      it 'aggregates per events' do
+        result = unique_count_service.per_event_aggregation
+
+        expect(result.event_aggregation).to eq([1])
+        expect(result.event_prorated_aggregation.map { |el| el.ceil(5) }).to eq([16.fdiv(31).ceil(5)])
+      end
+
+      context 'when removed on the last day of the period' do
+        let(:removed_at) { to_datetime }
+
+        it 'aggregates per events' do
+          result = unique_count_service.per_event_aggregation
+
+          expect(result.event_aggregation).to eq([1])
+          expect(result.event_prorated_aggregation).to eq([1])
+        end
+      end
+    end
+
+    context 'with persisted metrics added and removed in the period' do
+      let(:added_at) { from_datetime + 1.day }
+      let(:removed_at) { to_datetime - 1.day }
+
+      it 'aggregates per events' do
+        result = unique_count_service.per_event_aggregation
+
+        expect(result.event_aggregation).to eq([1])
+        expect(result.event_prorated_aggregation.map { |el| el.ceil(5) }).to eq([29.fdiv(31).ceil(5)])
+      end
+
+      context 'when added and removed the same day' do
+        let(:added_at) { from_datetime + 1.day }
+        let(:removed_at) { added_at }
+
+        it 'aggregates per events' do
+          result = unique_count_service.per_event_aggregation
+
+          expect(result.event_aggregation).to eq([1])
+          expect(result.event_prorated_aggregation.map { |el| el.ceil(5) }).to eq([1.fdiv(31).ceil(5)])
+        end
+      end
+    end
+
+    context 'with multiple events added in the period and with one added and removed during period' do
+      let(:added_at) { from_datetime + 10.days }
+      let(:quantified_event_2) do
```

Comment:
Naming/VariableNumber: Use normalcase for symbol numbers. (https://rubystyle.guide#snake-case-symbols-methods-vars-with-numbers)